alerting/src/main/kotlin/org/opensearch/alerting/util/IndexUtils.kt

```diff
@@ -26,6 +26,8 @@ import org.opensearch.index.IndexNotFoundException
 class IndexUtils {
 
     companion object {
+        val VALID_INDEX_NAME_REGEX = Regex("""^(?![_\-\+])(?!.*\.\.)[^\s,\\\/\*\?"<>|#:\.]{1,255}$""")
```

Comment:
Based on [this](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/metadata/IndexNameExpressionResolver.java#L1030), we just can't start with `_`.

Reply:
I had based this validation on [this function](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java#L267), and the documentation [here](https://opensearch.org/docs/latest/api-reference/index-apis/create-index/#index-naming-restrictions).